btrdbextras/opendss_ingest/simulation_utils.py

```diff
@@ -0,0 +1,569 @@
+                tags.append(tM)
+                tags.append(tA)
+                annotations.append(aM)
+                annotations.append(aA)
+
+    return collections, names, tags, annotations
+
+
+def get_existing_streams(col_prefix, conn):
+    """Get the existing streams under the base collection col_prefix"""
+    streams = conn.streams_in_collection(col_prefix)
+    # Build the dictionary of the streams
+    streams_dict = {}
+    for stream in streams:
+        streams_dict[stream.collection + "/" + stream.name] = stream
+    print("Found", len(streams_dict.keys()), "streams under", col_prefix)
+    return streams_dict
+
+
+def create_streams(
+    col_prefix: str,
+    collections: List,
+    names,
+    tags,
+    annotations,
+    conn: BTrDB,
+    verbose: bool = False,
+):
+    """
+    Given a set of collections, names, tags, and annotations for intended streams, check if
+    they exist. If not, create them.
+
+    Returns
+    -------
+    existing : dict
+        A dictionary capturing all the intended streams. Keys are the collection/name of the stream,
+        values are stream objects.
+    """
+
+    existing = get_existing_streams(col_prefix, conn)
+
+    # Iterate through the desired streams and check if they exist already. If not
+    # create them.
+    nstreams = len(collections)
+    nexisting = 0
+    ncreated = 0
+    for i in range(nstreams):
+        stream_info = collections[i] + "/" + names[i]
+        if stream_info in existing:
+            if verbose:
+                print(stream_info, "already exists.")
+            nexisting += 1
+            pass
+        else:
+            stream_id = uuid.uuid4()
+
+            stream = conn.create(
+                uuid=stream_id,
+                collection=collections[i],
+                tags=tags[i],
+                annotations=annotations[i],
+            )
+
+            existing[stream_info] = stream
+            if verbose:
+                print("Created", stream_info, ", uuid:", stream_id)
+            ncreated += 1
+    print("Found", nexisting, "streams. Created", ncreated, "streams.")
+    return existing
+
+
+def get_lineflow_stream_info(line_name, line_end, phase, ismag):
+    if ismag:
+        unit = "amps"
+    else:
+        unit = "degrees"
+    collection, name = get_lineflow_stream_colname(line_name, line_end, phase, ismag)
+
+    tags = {"name": name, "unit": unit}
+    annotations = {"phase": phase}
+
+    return collection, name, tags, annotations
+
+
+def get_lineflow_stream_colname(line_name, line_end, phase, ismag):
+    collection = line_name + "/" + line_end
+    if ismag:
+        lastltr = "M"
+    else:
+        lastltr = "A"
+
+    name = "I" + phase + lastltr
+    return collection, name
+
+
+def get_voltage_stream_column_name(bus_name, phase, ismag):
+    collection = bus_name
+    if ismag:
+        lastltr = "M"
+    else:
+        lastltr = "A"
+
+    name = "V" + phase + lastltr
+    return collection, name
```

Comment:
I think we can remove `bus_name` from this function. We return `collection` which is equal to `bus_name`.